Fix remaining TH6XXXWF services

bassdr opened the pull request:
I'll use this review to validate all services works compared to neviweb's configuration. Please do not merge before everything is checked (I'll edit them in-place). When it's checked, it's because I tried it and made sure Neviweb and the device took the option into account.

Feel free to test other devices are not broken too. Some changes might have affected other devices but I don't have them to validate.

# Configuration
 - [x] Temperature unit
 - [x] Time format
 - [x] Language
 - [x] Keypad
 - [x] Backlight
 - [x] Early start
 - [x] Filter change reminder
 - [x] Max. heating setpoint
 - [x] Min. heating setpoint
 - [x] Max. cooling setpoint
 - [x] Min. cooling setpoint
 - [x] Away heating setpoint
 - [x] Away cooling setpoint
 - [x] Do not allow heating if the outside temperature is above
 - [x] Do not allow cooling if the outside temperature si below
 - Eco Sinopé optimizations
     - [x] Heating optimization 
     - [x] Fan optimization
 
# Equipment
 - [x] Auxiliary heating source
 - [x] Reversing valve
 - [x] Accessories
 - [x] Heat dissipation time
 - [x] Cool dissipation time
 - [x] Heat Pump cycle length
 - [x] Aux. Cycle length
 - [x] Heat/cool setpoint delta
 - [x] Balance point
 - [x] Installation type
 - [x] Temperature calibration
 - [x] Compressor min. run time
 - [x] Compressor min. off time
 - [x] Auxiliary heating min. run time
 - [x] Auxiliary heating min. off time
 - [x] Min. activation delay between heating stages
 - [x] Min. activation delay between cooling stages
 - [x] Heat pump try time

Fixes #346 
Fixes #88 
Fixes #295 
Fixes #291 
Fixes #374 
Fixes #351

claudegel commented:
For `__init__.py` you put **device_id: str** but I can't test on my side as I have all my unique_id switched to string already.
As you also change in your setup self._id = str(device_info["id"]) you are at the same place then me. I hope it will not make a problem for the users that are still at numeric unique_id.

bassdr commented:
> you change self._id = device_info["id"] to self._id = str(device_info["id"])
> if we make that change all climate devices will be deactivated by HA and new devices will be added with a pending _2 at the end. This will break all automations for the user.

Are you ok if I attempt an automatic migration instead? The idea is to migrate existing users that has ints as unique_id to string in the registry, before adding the entities. This way it's best of both worlds: no breaking change, and we comply to the signature.

bassdr commented:
[c151f77](https://github.com/claudegel/sinope-130/pull/388/commits/c151f7792595089b2560b4554e98ffb1d6151860) is migrating the unique_id to str in th6xxxwf branch too, tested it many times (checkout master, trash everything, add a new device, then update), it works flawlessly for me.

claudegel commented:
Perfect after merging the th6xxxwf branch to master and making a new release then many people will be able to test neviweb130-V2 without having a problem to go back to master in case of problem. both will have str unique_id.
where is your fix for the TH1123ZB ? I don't see a commit for this, only the unique_id.

claudegel commented:
Can't see all your code but in the switch for int -> str, did you change the last line for each extra_state_attributes, climate, light, sensor, switch and valve
"id": str(self._id), -> "id": self._id,
In TH6xxxwf it's all "id": str(self._id),

bassdr commented:
> where is your fix for the TH1123ZB ? I don't see a commit for this, only the unique_id.

It's been a while, it's already merged in th6xxxwf: 06c650647ad0fcee9bd5c458bcc5aa3fca0942a5

Relates to #391 

Tho I must admin that without having an actual TH11123ZB, it's hard to say if it was the same issue, but I did have the same issue described on my TH6500WF (still problematic in master)
There's also a crash in master if you have the verbose logs enabled.

bassdr commented:
> Can't see all your code but in the switch for int -> str, did you change the last line for each extra_state_attributes, climate, light, sensor, switch and valve
> "id": str(self._id), -> "id": self._id,
> In TH6xxxwf it's all "id": str(self._id),

I updated every platform, yes. The most important part is in the `__init__`, where I convert `device["id"]` to str right away before setting `self._id`. I just updated the "id": self._id thing, it's cleaner and avoids a copy, thanks for pointing it out.

bassdr commented on custom_components/neviweb130/schema.py:
```diff
@@ -293,14 +294,14 @@
     {
         vol.Required(ATTR_ENTITY_ID): cv.entity_id,
-        vol.Required(ATTR_ROOM_SETPOINT_MAX): vol.All(vol.Coerce(float), vol.Range(min=8, max=36)),
+        vol.Required(ATTR_ROOM_SETPOINT_MAX): vol.All(vol.Coerce(float), vol.Range(min=11, max=30)),
```
@claudegel, do you know where those initial values were from? If another device can go as low as 8oC for the minimum, I can leave it to 8 and 36 for max but then raise an error if we're having unsupported values for the TH6500WF. What do you think?

bassdr commented on custom_components/neviweb130/schema.py:
```diff
@@ -293,14 +294,14 @@
     {
         vol.Required(ATTR_ENTITY_ID): cv.entity_id,
-        vol.Required(ATTR_ROOM_SETPOINT_MIN): vol.All(vol.Coerce(float), vol.Range(min=5, max=26)),
+        vol.Required(ATTR_ROOM_SETPOINT_MIN): vol.All(vol.Coerce(float), vol.Range(min=5, max=29)),
```
@claudegel, same here but the other way around, if another device can't go above 26 I'll raise an error for the other device.

bassdr commented on custom_components/neviweb130/schema.py:
```diff
@@ -345,14 +346,14 @@
     {
         vol.Required(ATTR_ENTITY_ID): cv.entity_id,
-        vol.Required(ATTR_COOL_SETPOINT_MAX): vol.All(vol.Coerce(float), vol.Range(min=16, max=30)),
+        vol.Required(ATTR_COOL_SETPOINT_MAX): vol.All(vol.Coerce(float), vol.Range(min=21, max=36)),
```
@claudegel, same here

bassdr commented:
I'm almost done testing the first section of setting (Configuration), but I'm not sure if the services already exist for "EcoSinope" thing...

- Eco Sinopé optimizations
    -  Heating optimization
    -  Fan optimization

It's two on/off switches. Is it the first device that has it or do we already have a service for something similar?

In the code, I see:
 - set_hvac_dr_options
 - set_hvac_dr_setpoint
 - set_load_dr_options

I guess they are not quite the same as what I need?
<img width="427" height="484" alt="image" src="https://github.com/user-attachments/assets/95c34372-085b-4501-a925-b8f6d7ee0062" />

claudegel commented:
There is a small pre-commit PR waiting but it fail because in the master it is still at 79 characters per lines. I can merge it unless you need to do something before. Can you have a look to the proposed changes and if it is ok I'll merge.
Also I just merge my PR to th6xxxwf and I took the time to put all constant and services in const.py in alphabetical order. It will be easier to add new constant without fooling around.
This fix support for TH1143WF and TH1144WF.

bassdr commented:
> I suppose that the first one is available only if you set aux heating as fossil energy or something like that.

No they did not do it that way (they should IMO). Instead, it's available as long as you have an auxiliary heater connected and configured (W1 or W2 pin detected at initial setup)

bassdr commented:
> Eco Sinopé optimisations is specific for the TH6xxx thermometers

I was able to create a service that works fine for TH6500WF. I modified the existing set_hvac_dr_options, hopefully this is what you meant, if not this is no problem, just tell me I'll create a new dedicated service. But I personally think we already have too many services so I didn't feel like adding another one.

claudegel commented:
For the unique_id migration I'm on str unique_id since many month and sometime I just forget to replace the str() before restart when I download the new version and I got many _2. But it is easy to fix. And I don't think we will have problem with that. Maybe we can add some doc in the readme in case somebody get that _2.
About the services, yes there are many but I just finish with V2 to port all of them to attribute entities and it work real fine. It is also extremely easy to create a new one if we need. I'll post a PR to V2 soon. I think that I'll be able to remove many services as user will prefer to do device configuration in the device page. We just need to keep some for backward compatibility to give user time to migrate there automation to the new entities. It's peak season that is coming. I should make blueprints to help them with new attributes entities.
I'll start to port the th6xxxwf to V2 this weekend.

claudegel commented:
In V2 I've added a session manager that close session before HA restart. This almost remove the ACCSESSEXC bug even if I restart every 5 minutes to test some changes. I've three simultaneous connection to Neviweb and sometime I need to go on Neviweb directly to check some parameters. So the 5 sessions limit is easy to bust.

bassdr commented:
> In V2 I've added a session manager that close session before HA restart. This almost remove the ACCSESSEXC bug even if I restart every 5 minutes to test some changes. I've three simultaneous connection to Neviweb and sometime I need to go on Neviweb directly to check some parameters. So the 5 sessions limit is easy to bust.

Perfect then, I'll leave my patch either way in here as it's still better, and we'll see what it looks like in V2 when we're at it. I'll try to go thru all advanced options today but I probably won't have time (unless they are all already working, but I doubt)

Realistically, I think this branch will be ready to merge (to master IMO) in about a week. We could then create a new version (I had in mind 4.0, because some services are not backward compatible), and then I'll go back to V2 (eventually version 5.0?) branch and help you finalize it. What do you think about this?

claudegel commented:
There is a small bug in valve.py self._flowmeter_opt_action = None in Neviweb130WifiValve should be move to class Neviweb130MeshValve. I'm porting al th6xxxwf to V2 and I'll make a small PR to th6xxxwf to fix what I find.

claudegel commented:
For set_time format service there is more then one class that do not have timeFormat. maybe we can change code like this ?
```
def set_time_format_service(service: ServiceCall) -> None:
    """Set time format 12h or 24h."""
    thermostat = get_thermostat(service)

    # Liste des classes d'appareils qui ne supportent pas le changement de format horaire
    unsupported_classes = (
        Neviweb130WifiLiteThermostat,
        Neviweb130HPThermostat,
        Neviweb130WifiHPThermostat,   <- that one is a new one that I need to add for HP6000WF-MA that just came out
    )

    if isinstance(thermostat, unsupported_classes):
        raise ServiceValidationError(
            f"Entity {thermostat.entity_id} does not support time format change"
        )

    value = {
        "id": thermostat.unique_id,
        ATTR_TIME: service.data[ATTR_TIME_FORMAT],
    }
    thermostat.set_time_format(value)
    thermostat.schedule_update_ha_state(True)

```

claudegel commented:
I've the changes for the HP6000WF-MA against th6xxxwf including all your present pr but if you prefer to finish th6xxwf and merge it before it's ok for me.

bassdr commented:
> Did you update to last HA version 2025.11.0 ? They remove the file home-assistant.log
> It's a very bad for debugging we have to go in parameters / system / log and it is extremely partial

No, I haven't, but are you sure you can't re-enable it with `hass --log-file LOG_FILE`?

This is the `hass --help` on 2025.10.4:
```
(sinope-130) david@bassdr ~/src/homeassistant/sinope-130 $ hass --help
usage: hass [-h] [--version] [-c path_to_config_dir] [--recovery-mode] [--debug] [--open-ui] [--skip-pip | --skip-pip-packages package_names] [-v] [--log-rotate-days LOG_ROTATE_DAYS] [--log-file LOG_FILE] [--log-no-color]
            [--script ...] [--ignore-os-check]

Home Assistant: Observe, Control, Automate.

options:
  -h, --help            show this help message and exit
  --version             show program's version number and exit
  -c, --config path_to_config_dir
                        Directory that contains the Home Assistant configuration
  --recovery-mode       Start Home Assistant in recovery mode
  --debug               Start Home Assistant in debug mode
  --open-ui             Open the webinterface in a browser
  --skip-pip            Skips pip install of required packages on startup
  --skip-pip-packages package_names
                        Skip pip install of specific packages on startup
  -v, --verbose         Enable verbose logging to file.
  --log-rotate-days LOG_ROTATE_DAYS
                        Enables daily log rotation and keeps up to the specified days
  --log-file LOG_FILE   Log file to write to. If not set, CONFIG/home-assistant.log is used
  --log-no-color        Disable color logs
  --script ...          Run one of the embedded scripts
  --ignore-os-check     Skips validation of operating system
  ```

bassdr commented:
> I still have few changes to th6xxxwf but it is against your th6xxxwf-fix-remaining. so maibe it is better to merge it in th6xxxwf then I'll push my changes along with the HP6000WF-MA.
> It's all ready for me

If it's already done then yes, don't hesitate to push it, and yes you can branch from th6xxxwf-fix-remaining too. I am lagging behind there are many services missing and it's a hard week for me... Changing position at work + I'm quite affected by the metro strike.

claudegel commented:
For the home-assistant.log in 2025.11.0 we can only view it in the console or in parameter / system / log but it is painfull. I've found this:
https://github.com/Petro31/a-file-logger
it bring back home-assistant.log so this gave me an idea for V2.
In config_flow I have the possibility to change the log level but also to drop all log pertinent to neviweb130 into a file in config, neviweb130_log.txt. This does not work anymore as home-assistant.log is gone but I can use the above file-logger to have my own log for just neviweb130.

claudegel commented:
I finally add a logging system in V2. I catch all logging for neviweb130 and write it to file config/neviweb130_log.txt. It work perfectly. I also added many config in config-flow to:
- set max log_file size, default to 2 meg
- set how many log rotation to manage, default 3
- set option to overwrite log_file or append to log_file on HA restart
- set dynamically log_level, default to INFO
- add a download log option that allow user to download the file from his browser to be able to send it for debug or just view it, in case they have not access to config.
- mask sensitive data in log
Neviweb130 will rotate automatically the log as it reach the file size limit.

bassdr commented:
> In config_flow I have the possibility to change the log level but also to drop all log pertinent to neviweb130 into a file in config, neviweb130_log.txt

I'm not a big fan of finding our ways around the decisions taken upstream. Let's read about why they removed this log and I'm quite sure the reason is that there are much better ways. I'm not using this log at all by the way, the errors get redirected to my debugger's console or I get them from docker logs, or from HomeAssistant's UI (depending on what I'm debugging)

Of course, I did not try 2025.11 so maybe I'll change my mind when trying it tho. I'm not planning on upgrading before it reaches 2025.11.2 or 2025.11.3, first versions of the month are usually unstable.
